libs/text-splitters/langchain_text_splitters/base.py

```diff
@@ -227,6 +231,7 @@ def __init__(
         model_name: Optional[str] = None,
         allowed_special: Union[Literal["all"], AbstractSet[str]] = set(),
         disallowed_special: Union[Literal["all"], Collection[str]] = "all",
+        tokenizer: Optional[Any] = None,
```

Comment:
The existing implementations aren't well documented which makes reviewing the changes much harder. I don't really know what some of the splitters were even designed to accept. Should TextSplitter or TokenTextSplitter even accept a tokenizer?

Is TokenTextSplitter really meant to only support tiktoken? Or should it support huggingface tokenizer as well? Are the parameterizations compatible?

cc @baskaryan in case you have more context

Reply:
Yeah that confused me as well. I feel like `from_huggingface_tokenizer` and `from_tiktoken_encoder` should  be class methods for `TokenTextSplitter`

Reply:
And TokenTextSplitter should support any support of tokenizer.

Reply:
I can document this.